modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2TreeIndexBase.java

```diff
@@ -101,13 +111,26 @@ static int computeInlineSize(List<InlineIndexColumn> inlineIdxs, int cfgInlineSi
         int size = 0;
 
         for (InlineIndexColumn idxHelper : inlineIdxs) {
-            if (idxHelper.size() <= 0) {
-                size = propSize;
-                break;
+            // for variable types - default variable size, for other types - type's size + byte type
+            int sizeInc = idxHelper.size() <= 0 ? IGNITE_VARIABLE_TYPE_DEFAULT_INDEX_SIZE : idxHelper.size() + 1;
+
+            if (idxHelper instanceof StringInlineIndexColumn) {
+                String sql = idxHelper.columnSql();
+
+                if (sql != null) {
+                    Matcher m = STRING_WITH_LENGTH_SQL_PATTERN.matcher(sql);
+
+                    if (m.find())
+                        // if column has defined length we use it as default
+                        sizeInc = Integer.parseInt(String.valueOf(m.group().subSequence(1, m.group().length() - 1)));
```

Comment:
Just realized that I lost my latest update. sizeInc should be equal to 
Integer.parseInt(String.valueOf(m.group().subSequence(1, m.group().length() - 1))) + 3
3 bytes should reserved by default. May be something similar should be done for string 115 for IGNITE_VARIABLE_TYPE_DEFAULT_INDEX_SIZE